sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonBenchmark.scala

```diff
@@ -86,6 +86,7 @@ object JSONBenchmark extends SqlBasedBenchmark {
         spark.read
           .schema(schema)
           .json(path.getAbsolutePath)
+          .filter((_: Row) => true)
```

Comment:
> This is another benchmark case, isn't it?

Originally I added the benchmark to check how specifying of `encoding` impacts on performance. This worked well till #21909 . Currently the benchmark just test how fast JSON datasource can create empty rows (in the case of `count()`) which is checked by another benchmark. 

I believe this PR is just follow up of #21909 which must include the changes proposed in the PR.

Reply:
@MaxGekk . In your PR, you already show the side effect via benchmark .

What I mean is both test cases are meaningful and worth to have. :) And, we need to compare both results in the future release.

In any way, please create new different benchmark cases for this PR.

Reply:
BTW, how do you want to call old and new test cases?
- For old case, we can give a new name.
- For new case, `JSON per-line parsing:` looks not a little bit accurate because we have filters now.